src/shogun/machine/Machine.cpp

```diff
@@ -176,3 +176,43 @@ std::shared_ptr<LatentLabels> Machine::apply_latent(std::shared_ptr<Features> da
 	error("This machine does not support apply_latent()");
 	return NULL;
 }
+
+std::shared_ptr<Machine> Machine::fit(
+    const std::shared_ptr<Features>& data, const std::shared_ptr<Labels>& lab)
+{
+	require(data, "No features provided");
```

Comment:
you need to store bore data and labels in the member here, that is the whole point of the class :)